Testing/SystemTests/tests/framework/DocumentationTest.py

```diff
@@ -0,0 +1,52 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from abc import ABCMeta, abstractmethod
+from pathlib import Path
+from typing import List
+
+from systemtesting import MantidSystemTest
+
+path_to_docs = (Path(__file__).resolve() / "../../../../../docs/source").resolve()
+
+
+class DocumentationTestBase(MantidSystemTest, metaclass=ABCMeta):
+    def __init__(self):
+        super(DocumentationTestBase, self).__init__()
+        self._known_no_docs: List[str] = []
+        self._docs_path: Path = Path("")
+        self._files: List[List[str]] = []
+        self._test_type: str = ""
+        self._setup_test()
+
+    @abstractmethod
+    def _setup_test(self):
+        raise NotImplementedError("This method needs implementing in a subclass of this class.")
+
+    def file_exists_under_docs_path(self, file_name: str) -> bool:
+        """
+        Search for file name in the given directory and all its subdirectories
+        :param file_name: name of the file
+        :return: bool, true if file found
+        """
+        found_files = [path for path in self._docs_path.rglob(file_name) if path.is_file]
+        return len(found_files) > 0
```

Comment:
It might be possible to simplify this function a bit further, would this work?
```suggestion
        document_path = self._docs_path / file_name
        return document_path.exists()
```

Reply:
The point is to look for the file anywhere within the docs path. So in the case of the algorithm that would work, but I think for things like the interfaces, there are lots of subdirectories to search. Maybe it's possible with a regex in a neater way?

Reply:
Ah right ok, in that case keep it as it is!